### ISSUE TITLE
Salt Master does not create pubkey signing key pair

### ISSUE THREAD

**Issue by et304383** [saltstack/salt#54238]
### Description of Issue
Salt master is not creating a keypair on startup as the documentation says it should when I set `master_sign_pubkey: True`

https://docs.saltstack.com/en/latest/topics/tutorials/multimaster_pki.html#prepping-the-master-to-sign-its-public-key

### Steps to Reproduce Issue
Install Ubuntu 18.04
Install salt-master via apt
Add `master_sign_pubkey: True` to configuration
Restart salt master

Expect to see keypair created in `/etc/salt/pki`

### Versions Report
```
Salt Version:
           Salt: 2017.7.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.2
         Python: 3.6.8 (default, Jan 14 2019, 11:02:34)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.2.5

System Versions:
           dist: Ubuntu 18.04 bionic
         locale: UTF-8
        machine: x86_64
        release: 4.15.0-1045-aws
         system: Linux
        version: Ubuntu 18.04 bionic
```

**Comment by frogunder** [comment]
@et304383 thank you for reporting this.

Seems like you are running an older version of salt which is currently CVE supported only.
Would you mind trying to install a newer version of salt and see if this is still an issue for you?
The current newest version of salt is 2019.2.0.
Please check out the SaltStack Package Repo for installation.
http://repo.saltstack.com/

Thanks

**Comment by et304383** [comment]
I installed the default version available via apt in Ubuntu 18.04.  It seems that version is out of date?

**Comment by frogunder** [comment]
@et304383 It seems that the version you get when doing `apt-install salt-master` on ubuntu18.04 is 2017.7.4. So seems that might not be updated, not sure what decides what version to get there.

If I do a `apt search salt` I see this 
```
salt-master/bionic-updates 2017.7.4+dfsg1-1ubuntu18.04.1 all
  remote manager to administer servers via salt

salt-minion/bionic-updates 2017.7.4+dfsg1-1ubuntu18.04.1 all
  client package for salt, the distributed remote execution system
``` 

I would suggest to use our repo http://repo.saltstack.com/ to install salt and preferably a supported version.
